### ISSUE TITLE
Date Range calendar

### ISSUE THREAD

**Issue by roberbnd** [ghiscoding/slickgrid-react#391]
### Describe the bug

I cant find a solution for show the calendar always inside of the view
if the column is in the edge, its happening this:
![image](https://github.com/user-attachments/assets/3fb07c1c-32fb-433a-96d5-666f4963a358)

### Reproduction

1. No click  date range filter
2. Move a column with dateRage filter to be the last column
3. click in the filter (dateRange)

---

**No do this**
when its no the last column and you click filter you are going to see:
![image](https://github.com/user-attachments/assets/1da0aaa4-1cf0-4288-8248-73d1a3b2480c)

and move the column to be de last is not happening the issue
![image](https://github.com/user-attachments/assets/446ef93d-0830-4291-bedf-6dfd9f5adf5c)

### Expectation

the calendar must be totally visible

### Environment Info

```shell
any browser
you can replicate it here
https://ghiscoding.github.io/slickgrid-react/#/example23
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/ghiscoding/slickgrid-react/blob/master/.github/CODE_OF_CONDUCT.md)
- [X] Read the [HOWTO - Step by Step](https://github.com/ghiscoding/slickgrid-react/wiki/HOWTO--Step-by-Step).
- [X] Check that there isn't [already an issue](https://github.com/ghiscoding/slickgrid-react/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/ghiscoding/slickgrid-react/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by ghiscoding** [comment]
That's not at all a problem with Slickgrid-React but rather a problem with the external libray Vanilla-Calendar-Pro, I was actually the person who added the auto-positioning in this Vanilla-Calendar-Pro [Pull Request](https://github.com/uvarov-frontend/vanilla-calendar-pro/pull/278/files#diff-502bf958a9a18a9b67b384db5b30377f911cd256fa94768e31c763802f41656b) but I don't remember if the space calculation is just for the viewport regardless of the scroll or if it calculates with the scroll position. I assume it was probably calculating without the scroll position. I guess the question would be, in your first print screen, does it have enough space to show it on the left of the input? 

Also note that I'm currently using a custom fork of [Vanilla-Calendar-Pro](https://github.com/uvarov-frontend/vanilla-calendar-pro) that I created and named [Vanilla-Calendar-Picker](https://github.com/ghiscoding/vanilla-calendar-picker). I'm trying to push all my changes to Vanilla-Calendar-Pro, so that I can switch back to that library and eventually delete my custom fork,.. so any fixes would probably have to be in [Vanilla-Calendar-Picker](https://github.com/ghiscoding/vanilla-calendar-picker) **not** in Slickgrid-React itself (it would however require a dependency update once a fix is found)

**Comment by roberbnd** [comment]
**does it have enough space to show it on the left of the input?**
No I haven't it, I was expecting to see the calendar but I have to scroll

I tried that configuration and I noticed this is happening
![image](https://github.com/user-attachments/assets/2760a0b8-0a7f-41ae-b720-21f59f2fc1cf)

then when you want to see the calendar for value ['bottom', 'left'] you going to see
![image](https://github.com/user-attachments/assets/90b25ffe-ddc1-42f6-bdb8-c4f2325ccf62)


and I was expecting 
![image](https://github.com/user-attachments/assets/17954864-5d70-4239-a7ab-373cd1a2e790)



My first approach was add negative margin to move the calendar
```css
.vanilla-calendar {
  margin-left: -20px
} 
```

but in the end I created the same issue in the right of the screen

**Comment by roberbnd** [comment]
I was reviewing your code I think that I get it, I issue is happening for the scroll
Example here
![image](https://github.com/user-attachments/assets/626e17fb-629e-4f85-bbf5-9535d9036141)
as you can see I have 2 scrolls 
1.-the common scroll 
2.-the slickgrid scroll


the calculation only has in consideration the first scroll its missing the scroll of the slickgrid
![image](https://github.com/user-attachments/assets/87787803-96f8-4a59-a75c-45a3bda72a5c)

**Comment by roberbnd** [comment]
for ['bottom', 'right'] I got this:
![image](https://github.com/user-attachments/assets/1d6ef284-9d8f-4870-846f-76ed4ad38660)

**Comment by ghiscoding** [comment]
well it's bottom-left then, but it doesn't seem to be correct. However I didn't work on that part, the original author of the lib did that, not me and the `auto` position won't help either

**Comment by roberbnd** [comment]
anyway thanks! now I understand what is happening.

**Comment by ghiscoding** [comment]
the author used Tailwind in his project and I never used Tailwind before, but if you look at the CSS theme, it's probably these lines that makes the calendar stretch that way

https://github.com/ghiscoding/vanilla-calendar-pro/blob/7301d4468bf2c512adca08d5c3cccdfe2203e059/package/src/styles/vanilla-calendar.layout.css#L11-L27

I assume it's probably using Flexbox and it is what makes the calendar show like your first print screen, you are right it's a bit weird how it shows but it's really not my code so... I will close this issue since there is nothing to do here, the problem would have to be fixed in the external library, not Slickgrid-React, so closing.

**Comment by ghiscoding** [comment]
Ok so after investigating a bit, I found the problem is because the Calendar Picker is appended to the `document.body` and then it calculates the position of the picker by reading the input x,y position, but the problem is that the input position is no longer in the visible viewport but we use these x,y coordinates to change the position of the calendar picker, but this picker is appended to the body **not** to the grid div (that we need to scroll to see). So because the picker is on the body, it tries to use a x,y that is too much to the right, however it doesn't allow it to go further because it's on the body, which is why it doesn't follow the input (because the input is with the grid div, and the picker is on the body, so they are both in different DOM layers).

Also the reason why the calendar picker becomes very tall with both months showing as top and bottom is because it uses Flexbox and because it's appended to the body and it cannot go wider than the body viewport. Flexbox are great when it has enough space, but when it does then it tries to stretch to find enough space to show its content, so it's not always good to use Flexbox

So this issue should be opened on the external project [Vanilla-Calendar-Pro](https://github.com/uvarov-frontend/vanilla-calendar-pro), not here, which is why I closed the issue.

**Comment by ghiscoding** [comment]
@roberbnd ok so I managed to fix the issue in my fork that I just mentioned above in previous [comment](https://github.com/ghiscoding/slickgrid-react/issues/391#issuecomment-2292804899), you can see this [PR](https://github.com/ghiscoding/vanilla-calendar-picker/pull/7) to fix that problem. It doesn't fix all your problem but at least it will not show the picker with a month on the top and bottom

you can see with the animated gif below that it helps a lot, it keeps the correct size of the picker because I now detect if the picker new coordinate is outside the document body viewport, I added a condition check for that and if it's outside then I reposition the picker to be just beside the body viewport (it no longer follows the input position when it cannot)

![brave_QdIrb7TJFD](https://github.com/user-attachments/assets/e46cb301-82b2-4a59-9ed2-44d977ed7ff0)

Very Important... to get this fix, you will need to update the Vanilla-Calendar-Picker (that is my project fork of that library), I just released a new version with this fix 
https://www.npmjs.com/package/vanilla-calendar-picker?activeTab=versions

To update the dependency, you will probably need to delete your lock file (npm, yarn or pnpm lock file) and then run npm install (or other) and it should pick up the latest version. I will wait a few more days or week before pushing a version of Slickgrid-Universal, so deleting your lock file is the easiest way to get this fix.

**Comment by ghiscoding** [comment]
pushed a new release for another bug fix, so it's now also including the Vanilla Calendar fixes [v5.5.1](https://github.com/ghiscoding/slickgrid-react/releases/tag/v5.5.1), cheers ⭐
